File: workout-tracker-app-webservice/backend/src/main/java/hu/bme/aut/workout_tracker_backend/data_layer/saved_exercise/SavedExercise.java
package hu.bme.aut.workout_tracker_backend.data_layer.saved_exercise;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class SavedExercise {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO, generator="native")
    @GenericGenerator(name = "native")
    private Long id;
    private String userId;
    private Long exerciseId;
    @ElementCollection
    private List<String> data = new ArrayList<>();
}

File: workout-tracker-app-webservice/app/app/src/main/java/hu/bme/aut/workout_tracker/data/model2/SavedExercise.kt
package hu.bme.aut.workout_tracker.data.model2


import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class SavedExercise(
    @Json(name = "id")
    val id: Int,
    @Json(name = "userId")
    val userId: String,
    @Json(name = "exercise")
    val exercise: Exercise,
    @Json(name = "data")
    val data: ArrayList<String>
)